src/protocol/src/lib.rs

```diff
@@ -147,6 +159,64 @@ pub enum Command {
     Terminate,
 }
 
+/// Manages allocation of Connection IDs and secret keys.
+pub struct ConnSupervisor {
+    next_id: i32,
+    max_id: i32,
+    free_ids: VecDeque<i32>,
+    current_mapping: HashMap<i32, i32>,
+}
```

Comment:
Can you please use type aliases `ConnId` and `ConnSecretKey` respectively. Also in function signature respectively too. Thanks